hedera-cryptography/hedera-cryptography-tss/src/main/java/com/hedera/cryptography/tss/api/TssService.java

```diff
@@ -0,0 +1,167 @@
+ */
+public interface TssService {
+
+    /**
+     * Generate a TSS message for a pendingParticipantDirectory, from a random private share.
+     *
+     * @param pendingParticipantDirectory the pending participant directory that we should generate the message for
+     * @return a TSSMessage produced out of a random share.
+     */
+    @NonNull
+    TssMessage generateTssMessage(@NonNull TssParticipantDirectory pendingParticipantDirectory);
+
+    /**
+     * Generate a TSS message for a pendingParticipantDirectory, for the specified {@link TssPrivateShare}.
+     *
+     * @param pendingParticipantDirectory the pending participant directory that we should generate the message for
+     * @param privateShare       the secret to use for generating new keys
+     * @return a TSSMessage for each owned share.
+     */
+    @NonNull
+    TssMessage generateTssMessages(
+            @NonNull TssParticipantDirectory pendingParticipantDirectory, @NonNull TssPrivateShare privateShare);
+
+    /**
+     * Verify that a {@link TssMessage} is valid.
+     *
+     * @param participantDirectory the participant directory used to generate the message
+     * @param tssMessages the TSS messages to validate
+     * @return true if the message is valid, false otherwise
+     */
+    boolean verifyTssMessage(@NonNull TssParticipantDirectory participantDirectory, @NonNull TssMessage tssMessages);
+
+    /**
+     * Compute all private shares that belongs to this participant from a threshold minimum number of {@link TssMessage}s.
+     * It is the responsibility of the caller to ensure that the list of validTssMessages meets the required threshold.
+     *
+     * @param participantDirectory the pending participant directory that we should generate the private share for
+     * @param validTssMessages the TSS messages to extract the private shares from. They must be previously validated.
+     * @return a sorted by sharedId list of private shares the current participant owns, or null if there aren't enough shares to meet the threshold.
+     */
+    @Nullable
+    List<TssPrivateShare> decryptPrivateShares(
+            @NonNull TssParticipantDirectory participantDirectory, @NonNull List<TssMessage> validTssMessages);
+
+    /**
+     * Aggregate a threshold number of {@link TssPrivateShare}s.
+     * It is the responsibility of the caller to ensure that the list of private shares meets the required threshold.
+     * If the threshold is not met, the private key returned by this method will be invalid.
+     *
+     * @param privateShares the private shares to aggregate
+     * @return the aggregate private key
+     */
+    @NonNull
+    PairingPrivateKey aggregatePrivateShares(@NonNull List<TssPrivateShare> privateShares);
+
+    /**
+     * Compute all public shares for all the participants in the scheme.
+     *
+     * @param participantDirectory the pending participant directory that we should generate the message for
```

Comment:
this doc mentions generating messages, yet the method isn't generating any messages